pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -249,6 +253,37 @@ var (
 	RotationPolicyAlways PrivateKeyRotationPolicy = "Always"
 )
 
+// The type of additional private key format to write
+// Allowed values are `DER` or `PEM`.
+// When Type is set to `DER` an additional entry `key.der` will be created in the secret
+// containing the binary format of the key.
+// When Type is set to `CombinedPEM` and additional entry `tls.pem` will be created in the secret
+// containing the PEM formated certificate (tls.key + tls.crt concatened)
+// +kubebuilder:validation:Enum=DER;CombinedPEM
+type OutputFormatType string
+
+const (
+	// OutputFormatDER is the binary format of the key
+	// stored in `key.der` secret key
+	OutputFormatDER OutputFormatType = "DER"
+
+	// OutputFormatDERKey is the name of the data entry in the Secret resource
+	// used to store the DER formated private key
+	OutputFormatDERKey string = "key.der"
+
+	// OutputFormatPEM is a file containing both tls.crt and tls.key
+	// stored in `tls.pem` secret key
+	OutputFormatPEM OutputFormatType = "CombinedPEM"
+
+	// OutputFormatPEMKey is the name of the data entry in the Secret resource
+	// used to store the Combined PEM certifcate + key
+	OutputFormatPEMKey string = "tls.pem"
```

Comment:
suggestion: I feel like `tls.pem` really needs a different name. Both `tls.crt` and `tls.key` in a k8s TLS secret contain a PEM-encoded value, so `tls.pem` could work as a name for both of them. It runs the risk of being confusing.

I'd prefer something verbose but more clear such as `tls-combined.pem` or `tls-chain-key-combined.pem`. I'm not _thrilled_ with either of those suggestions, but I really worry that `tls.pem` isn't clear enough.

Reply:
OK, i change to `tls-combined.pem`